pkg/controller/resourcequota/resource_quota_controller.go

```diff
@@ -339,100 +294,38 @@ func (rq *ResourceQuotaController) syncResourceQuota(quota api.ResourceQuota) (e
-		sum = &q
-	}
-	return sum, nil
-}
-
-// PodHasRequests verifies that each container in the pod has an explicit request that is non-zero for a named resource
-func PodHasRequests(pod *api.Pod, resourceName api.ResourceName) bool {
-	for j := range pod.Spec.Containers {
-		value, valueSet := pod.Spec.Containers[j].Resources.Requests[resourceName]
-		if !valueSet || value.Value() == int64(0) {
-			return false
-		}
-	}
-	return true
-}
-
-// When a pod is deleted, enqueue the quota that manages the pod and update its expectations.
-// obj could be an *api.Pod, or a DeletionFinalStateUnknown marker item.
-func (rq *ResourceQuotaController) deletePod(obj interface{}) {
-	pod, ok := obj.(*api.Pod)
-	// When a delete is dropped, the relist will notice a pod in the store not
-	// in the list, leading to the insertion of a tombstone object which contains
-	// the deleted key/value. Note that this value might be stale. If the pod
-	// changed labels the new rc will not be woken up till the periodic resync.
-	if !ok {
+// When an object is deleted, enqueue the quota that manages the object.
+// obj could be a runtime.Object or a DeletionFinalStateUnknown marker item.
+func (rq *ResourceQuotaController) observeDelete(obj interface{}) {
+	metaObject, err := meta.Accessor(obj)
+	if err != nil {
 		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
 		if !ok {
-			glog.Errorf("Couldn't get object from tombstone %+v, could take up to %v before a quota records the deletion", obj, rq.resyncPeriod())
+			glog.Errorf("quota controller could not get object from tombstone %+v, could take up to %v before a quota usage reflects the deletion", obj, rq.resyncPeriod())
 			return
 		}
-		pod, ok = tombstone.Obj.(*api.Pod)
-		if !ok {
-			glog.Errorf("Tombstone contained object that is not a pod %+v, could take up to %v before quota records the deletion", obj, rq.resyncPeriod())
+		metaObject, err = meta.Accessor(tombstone.Obj)
+		if err != nil {
+			glog.Errorf("quota controller tombstone contained object that is not a meta %+v, could take up to %v before quota records the deletion", tombstone.Obj, rq.resyncPeriod())
 			return
 		}
 	}
 
-	quotas, err := rq.rqIndexer.Index("namespace", pod)
+	// look up the quota documents that are impacted
+	namespace := metaObject.GetNamespace()
+	indexKey := &api.ResourceQuota{}
+	indexKey.Namespace = namespace
+	quotas, err := rq.rqIndexer.Index("namespace", indexKey)
+
 	if err != nil {
-		glog.Errorf("Couldn't find resource quota associated with pod %+v, could take up to %v before a quota records the deletion", obj, rq.resyncPeriod())
+		glog.Errorf("quota controller could not find ResourceQuota associated with namespace: %s, could take up to %v before a quota records the deletion", namespace, rq.resyncPeriod())
 	}
+
 	if len(quotas) == 0 {
-		glog.V(4).Infof("No resource quota associated with namespace %q", pod.Namespace)
+		glog.V(4).Infof("No resource quota associated with namespace %q", namespace)
 		return
 	}
```

Comment:
Most likely I am misunderstanding it, but wondering if you are returning here, shouldn't it return after err != nil above too?